packages/calypso-e2e/src/lib/pages/media-page.ts

```diff
@@ -0,0 +1,148 @@
+	 *
+	 * If the media gallery has been filtered (eg. Images only), this method will select the
+	 * `nth` item in the filtered gallery.
+	 *
+	 * @param {[key: string]: number } param0 Parameter object.
+	 * @param {number} param0.item nth media gallery item to be selected.
+	 * @returns {Promise<void>} No return value.
+	 * @throws {Error} If requested item could not be located in the gallery. Alternatively,
+	 * if the click action failed to select the gallery item.
+	 */
+	async clickOn( { item }: { item: number } ): Promise< void > {
+		// Playwright is able to select the nth matching item given a selector.
+		// See https://playwright.dev/docs/selectors#pick-n-th-match-from-the-query-result.
+		const element = await this.page.waitForSelector(
+			`:nth-match(${ selectors.items }, ${ item })`
+		);
+
+		if ( ! element ) {
+			throw new Error(
+				`Requested item number ${ item } is greater than number of items in gallery.`
+			);
+		}
+
+		await element.waitForElementState( 'visible' );
+		await element.click();
+		await element.waitForElementState( 'stable' );
+		const isSelected = await element.evaluate( ( node ) =>
+			node.classList.contains( 'is-selected' )
+		);
+		if ( ! isSelected ) {
+			throw new Error( `Failed to select requested item number ${ item }` );
+		}
+	}
+
+	/**
+	 * Clicks on the navigation tab.
+	 *
+	 * @param {[key: string]: string } param0 Parameter object.
+	 * @param {string} param0.name Name of the tab to click.
+	 * @returns {Promise<void>} No return value.
+	 */
+	async clickTab( {
+		name,
+	}: {
+		name: 'All' | 'Images' | 'Documents' | 'Videos' | 'Audio';
+	} ): Promise< void > {
+		await this.page.waitForSelector( selectors.navtabs );
+		await this.page.click( `${ selectors.navtabs } span:has-text("${ name }")` );
+		// Wait for all placeholders to disappear.
+		// Alternatively, waiting for `networkidle` will achieve the same objective
+		// at the cost of much longer resolving time (~20s).
+		await this.page.waitForSelector( '.is-placeholder', { state: 'hidden' } );
+	}
+
+	/**
+	 * Given that a gallery item is selected, enter the edit screen.
+	 *
+	 * @returns {Promise<void>} No return value.
+	 */
+	async editImage(): Promise< void > {
+		await this.page.click( selectors.editButton );
+		await this.page.waitForSelector( selectors.mediaModalPreview );
+		await this.page.click( selectors.mediaModalEditButton );
+		await this.page.waitForSelector( selectors.imageEditorCanvas );
+	}
+
+	/**
+	 * Rotates the image.
+	 *
+	 * @returns {Promise<void>} No return value.
+	 */
+	async rotateImage(): Promise< void > {
+		const preview = await this.page.waitForSelector( selectors.imageEditorCanvas );
+		const selector = `${ selectors.imageEditorToolbarButton } span:text("Rotate")`;
+		await this.page.click( selector );
+		await preview.waitForElementState( 'stable' );
+		const undoButton = await this.page.waitForSelector( selectors.imageEditorResetButton );
+		await undoButton.isEnabled();
```

Comment:
This is a potential false negative as we will not know whether the button was eventually enabled or not. This will not throw if the button is not enabled, only return a boolean.
```suggestion
		await undoButton.waitForElementState( 'enabled' );
```

Reply:
I should mention here that we need to be careful when using this check, whether it's the `isEnabled()` or `waitForElementState( 'enabled' )`. It's because i.e. in Gutenberg, we've started moving away from the disabled attribute due to [some accessibility issues](https://github.com/Automattic/wp-calypso/pull/50911#discussion_r595871737). Instead the `disabled` attribute, the `aria-disabled` might be applied to indicate the element's state which is not covered by mentioned Playwright's methods. The one element that I know is using this convention is the editor's `Publish` button:

<img width="929" alt="Screenshot 2021-06-17 at 17 13 40" src="https://user-images.githubusercontent.com/1451471/122424950-6b282c80-cf8f-11eb-985a-166754737e92.png">

Reply:
Interesting to know what's happening behind the scenes, thanks for the heads up!

It is possible to check for the aria value using this:
```
const ariaValue = await testUndoButton.getAttribute('aria-disabled');
```

I ran this against the Gutenberg publish button in its disabled state, and the value of `ariaValue` became `true`. So going forward, we can check for `aria-disabled` attribute as this change rolls out.

For now though, the Image Editor's `Reset` button doesn't seem to have the `aria-disabled` attr yet, so I will accept your suggestion for the time being and perhaps explore writing our own `isDisabled` check in `browser-helper` or `browser-manager`?

Reply:
> So going forward, we can check for aria-disabled attribute as this change rolls out.

IMO, the biggest danger is that when an element is switched to use `aria-disabled`, we might not notice it because to Playwright, the button will always be enabled then. I thought we would be able to avoid it with Playwright, but you're right - we probably need a custom helper for that. Something like this should do the trick:

```js
/**
 * element-helper.js
 */

export async function waitForElementEnabled( elementHandle, options ) {
	return await this.page.waitForFunction( async () => {
		const isEnabled = await elementHandle.isEnabled();
		const isNotAriaDisabled = await elementHandle
			.getAttribute( 'aria-disabled' )
			.then( ( v ) => v !== 'true' );

		return isEnabled && isNotAriaDisabled ? elementHandle : null;
	}, options );
}
```

Maybe we should also create a linter rule that warns about using the native `waitForElementState( 'enabled' )`? What do you think?

Reply:
Sorry if I was not clear enough.

Playwright does indeed support querying aria attributes. All you need to do to check for `aria-disabled` is this:

`const ariaValue = await testUndoButton.getAttribute('aria-disabled');`

and it will return the appropriate value that is set for that attribute. 

> Maybe we should also create a linter rule that warns about using the native waitForElementState( 'enabled' )? What do you think?

I am inclined to pass on this at the moment, because I do not have a good sense of how quickly the project to migrate from `disabled` to `aria-disabled` is happening, or if there even is a project for this.

In my opinion, if an existing e2e test is broken by a PR where the changeover is taking place, then it should fall under the purview of the engineer making the change to placate the test. If they are unsure, they are free to reach out to the Quality Squad and we are happy to help. But otherwise, I do not want to prescribe what notion of `enabled` should be used.

This is assuming the migration to use `aria-disabled` is an ad-hoc one. If there is a concentrated effort to migrate, then I think a rule can possibly make sense, but at the end of the migration.

Reply:
> In my opinion, if an existing e2e test is broken by a PR where the changeover is taking place, then it should fall under the purview of the engineer making the change to placate the test.

Agreed! Just to be sure I'm clear myself, I meant that in some cases (like with the `undoButton`), if the element has been migrated to use the `aria-disabled` attribute, the test **won't fail**, because `waitForElementState( "enabled" )` will always return true as the `disabled` attribute will not be present at all. This is how we'd end up with a false-negative test result, which could go unnoticed for a very long time.

Reply:
@WunderBart Okay, I think I understand what you are saying.

I will look into implementing a `ElementHelper.waitForElementEnabled` and use that in this section of the code.

Reply:
@WunderBart I have implemented a custom function, `ElementHelper.waitForElementEnabled` which will wait for the element to become 'enabled' within a timeout value.

The helper function `ElementHelper.isElementEnabled` performs the job of checking both the disable property and aria-disabled attribute. 

Note, the `waitForFunction` (and other related functions like `evaluate`) runs in the page context and functions must use JS bindings, not Playwright calls.

Reply:
See https://github.com/Automattic/wp-calypso/pull/53724#discussion_r657113277